src/shared/components/button/RowLayout.tsx

```diff
@@ -54,153 +23,214 @@ type RowLayoutProps = React.ComponentPropsWithRef<typeof StyledRow> & {
-                    },
-                },
-                errorFree: {
-                    borderRadius: '20px',
-                    backgroundColor: `${theme.colors.inputBackground}`,
-                    boxShadow: '0px 1px 4px 0px rgba(0, 0, 0, 0.05)',
-                    border: `1px solid ${theme.colors.primary}`,
-
-                    '&:focus-visible': isFirefox
-                        ? {
-                              'outline-style': 'solid',
-                              'outline-width': '2px',
-                              'outline-offset': '-2px',
-                          }
-                        : {},
-                },
-            },
-        })};
-`;
-
 export const RowLayout = forwardRef(function RowLayout(
     {
+        address,
+        as,
+        children,
+        className,
+        currency,
+        disabled,
+        helperText,
+        iconClassName,
         iconLeading,
         iconTrailing,
-        title,
-        helperText,
+        ledgerIndicator,
+        onClick,
+        onKeyDown,
         rightHelperText,
-        children,
+        tabIndex = 0,
         testnetIndicator,
-        ledgerIndicator,
+        title,
         variant = 'primary',
-        disabled,
-        currency,
-        address,
-        tabIndex = 0,
-        as,
-        iconClassName,
-        ...rest
     }: RowLayoutProps,
     forwardedRef: React.Ref<HTMLDivElement>,
 ) {
-    const containerAs = as === 'button' ? 'span' : undefined;
+    const hasPointer = onClick !== undefined;
+
+    const containerStyles = cn(
+        'relative flex w-full max-h-[74px] p-4 gap-3 disabled:cursor-not-allowed disabled:pointer-events-none',
+        {
+            'cursor-pointer': hasPointer,
+            'cursor-auto': !hasPointer,
+            'border-none bg-transparent': as === 'button',
+            'rounded-2xl bg-white dark:bg-subtle-black shadow-[0_1px_4px_0_rgba(0,0,0,0.05)] hover:shadow-[0_0_0_1px] hover:shadow-theme-secondary-200 hover:dark:shadow-theme-secondary-600':
+                variant === 'primary',
+            'rounded-[20px] bg-white dark:bg-subtle-black shadow-[0_1px_4px_0_rgba(0,0,0,0.05)] border border-solid border-theme-primary-700 dark:border-theme-primary-650':
+                variant === 'errorFree',
+            'focus-visible:outline focus-visible:outline-2 focus-visible:-outline-offset-2':
+                isFirefox,
+        },
+        className,
+    );
+
+    if (as === 'button') {
+        return (
+            <button
+                className={containerStyles}
+                tabIndex={tabIndex}
+                ref={forwardRef as any as LegacyRef<HTMLButtonElement>}
+                onClick={onClick as any as MouseEventHandler<HTMLButtonElement>}
+                onKeyDown={onKeyDown as any as React.KeyboardEventHandler<HTMLButtonElement>}
+            >
+                <span className='w-full gap-3 flex items-star'>
```

Comment:
I'd turn all this into a single element with `<span>` (like this `button` implementation has) and not bother with a second variant that has `<div>` element with the same data in it. Instead you can move it out to a separate method if you want to be able to have both a `button` and `div` wrapper around it, and have both call the method for rendering the elements inside it